components/license-control-service/pkg/server/server.go

```diff
@@ -116,6 +116,22 @@ func (s *LicenseControlServer) Status(ctx context.Context, req *lc.StatusRequest
 		if err != nil {
 			return nil, status.Errorf(codes.Internal, "Deployment id and type not available: %s", err.Error())
 		}
+
+		//calculate Grace period status
+		gracePeriod := false
+		licensedEndTime := time.Unix(licensedPeriod.end.Seconds, 0)
+		if lic.Type == "commercial" {
+			if licensedEndTime.Before(time.Now()) {
+				//adding 30 days of grace period if license got expired
+				gracePeriodDate := time.Unix(licensedPeriod.end.Seconds, 0).AddDate(0, 0, 30)
+				//checks if current date and time is within the grace period
+				if time.Now().Before(gracePeriodDate) {
+					gracePeriod = true
+					licensedEndTime = gracePeriodDate
+				}
```

Comment:
Why we are making licensedEndTime to grace period date?

Reply:
Setting licensedEndTime to the grace period date allows the system to consistently recognize the license as valid during the grace period, ensuring that commercial license users have additional time to renew without losing access to services immediately upon license expiry.

Reply:
We don't want this, grace period only signifies if grace period is active or not.